config/src/network_id.rs

```diff
@@ -81,12 +81,69 @@ impl NetworkContext {
 /// and handshakes should verify that the NetworkId being used is the same during a handshake,
 /// to effectively ensure communication is restricted to a network.  Network should be checked that
 /// it is not the `DEFAULT_NETWORK`
-#[derive(Clone, Deserialize, Eq, Hash, PartialEq, Serialize)]
-#[serde(rename = "NetworkId", rename_all = "snake_case")]
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
 pub enum NetworkId {
     Validator,
     Public,
-    Private(String),
+    /// An invalid type due to migration from old NetworkId type to new type.  This will *not* deserialize
+    /// TODO: Once fully migrated, this could become Vfn through migration
+    Invalid,
+    Vfn,
+}
+
+impl Serialize for NetworkId {
```

Comment:
minor nit: i would add a comment here about the migration / conversion

Reply:
There is the comment on line 108